src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -267,10 +269,16 @@ JsonRpcResult GetParsingError(string error, Exception? exception = null)
             {
                 reader.AdvanceTo(buffer.End);
             }
-        }
 
-        // Completes the PipeReader's asynchronous reading operation.
-        await reader.CompleteAsync();
+            // Completes the PipeReader's asynchronous reading operation.
+            await reader.CompleteAsync();
+
+            if (isNewPayload)
+            {
+                Metrics.JsonRpcNewPayloadTime = stopwatch.ElapsedMilliseconds;
```

Comment:
If we want to add something lets add a dictionary and measure all the different types of RPC calls?

Probably would need to somehow aggregate multiple calls?